cmake/kokkos_compiler_id.cmake

```
@@ -112,7 +112,7 @@ IF(KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
   ENDIF()
   # The clang based Intel compiler reports as Clang to most versions of CMake
   EXECUTE_PROCESS(COMMAND ${CMAKE_CXX_COMPILER} --version
-                  COMMAND grep -c icpx
+                  COMMAND grep -c "DPC++\\|icpx"
```

I would prefer
```suggestion
                  COMMAND grep -c "\(DPC++\|icpx\)"
```
for readability.